src/main/java/org/apache/commons/io/FilenameUtils.java

```diff
@@ -201,7 +201,7 @@ public static String concat(final String basePath, final String fullFileNameToAd
         if (prefix < 0) {
             return null;
         }
-        if (prefix > 0) {
+        if (prefix > 0 && !fullFileNameToAdd.isEmpty() && '~' != (fullFileNameToAdd.charAt(0))) {
```

Comment:
Would this work too?

```suggestion
        if (prefix > 0 && !fullFileNameToAdd.startsWith("~")) {
```

Reply:
(I think the `startsWith` handles case of empty strings.

Reply:
@kinow @garydgregory Thank you for the feedback, I will test out the simplified change tonight and update the PR accordingly.